datachannel.go

```diff
@@ -182,12 +182,12 @@ func (d *DataChannel) open(sctpTransport *SCTPTransport) error {
 
 	dc, err := datachannel.Dial(d.sctpTransport.association, *d.ID, cfg)
 	if err != nil {
-		d.mu.RUnlock()
+		d.mu.Unlock()
 		return err
 	}
 
 	d.ReadyState = DataChannelStateOpen
```

Comment:
Can you make this so we just take the writelock for setting `d.ReadyState` I just don't want to be wasteful having a write lock for the whole function. If this is a bad idea please push back!

```
d.mu.RUnlock()
d.mu.Lock()
d.ReadyState = DataChannelStateOpen
d.mu.Unlock()
```

Reply:
I think, it wouldn't give us much because:
1. This method would virtually 100% succeed in production use (bug-free).
1. Sticking with only one kind of lock is relatively less error-prone.
1. DataChannel.open() won't be called that often anyway (I wouldn't worry)

Reply:
A agree with all that, I was overthinking it. More complicated code for no real gain :)